AppCenterData/AppCenterData/MSData.m

```diff
@@ -823,6 +826,40 @@ - (void)authTokenContext:(MSAuthTokenContext *)__unused authTokenContext didUpda
   }
 }
 
+#pragma mark - MSServiceNotificationDelegate
+
+- (void)appCenter:(MSAppCenter *)__unused appCenter didReceiveServiceNotification:(NSDictionary<NSString *, NSString *> *)notificationData {
+  NSString *documentChangePayloadJSON = notificationData[kMSServiceNotificationDocumentChangeKey];
+  if (documentChangePayloadJSON) {
+    NSData *documentChangePayloadData = [documentChangePayloadJSON dataUsingEncoding:NSUTF8StringEncoding];
+    NSError *error = nil;
+    NSArray *documentChangePayload = [NSJSONSerialization JSONObjectWithData:documentChangePayloadData
+                                                                     options:NSJSONReadingMutableContainers
+                                                                       error:&error];
+    if (error) {
+      MSLogError([MSData logTag], @"Couldn't parse json data for service notification: %@", error.localizedDescription);
```

Comment:
Please do not nest `if` statements. You can simply return here and remove at least one `else` statement below.

Reply:
I won't argue about methods having multiple return points vs nesting :) 

I have:
1. Added a return to remove the else block (as suggested)
1. Changed the check of documentChanges count to `== 0` and added a LogDebug before returning to reduce another level of nesting. (Open to suggestions on different log level - this is a no-op, but may indicate something odd going on in the backend so could maybe go higher - thoughts?)
1. Combined check of strongDelegate and `[strongDelegate respondsToSelector...` to remove another level of nesting.